src/v/cluster/partition_balancer_planner.cc

```diff
@@ -2149,6 +2156,124 @@ partition_balancer_planner::get_force_repair_actions(request_context& ctx) {
     }
 }
 
+absl::flat_hash_set<model::node_id>
+partition_balancer_planner::do_get_auto_decommission_actions(
+  const do_get_auto_decommission_actions_params& params) noexcept {
+    // rule: a quorum of nodes in the cluster must vote that a node has passed
+    // the current auto decommission timeout
+    size_t quorum = params.cluster_members.size() / 2 + 1;
+
+    // given a map from the health report on node_id -> decom_report, filter to
+    // 1. only cluster members
+    // 2. only reports that have a value
+    // 3. only reports that have the current configuration version
+    //    - this checks that we are working with the current decommission
+    //      timeout
```

Comment:
oh I see the purpose of config_version here.. I'm wondering if this is really needed. Can each broker just report the time_since_last_hb of every other broker and the controller makes a decision based based on the (latest) config version it sees (eliminates all this config_i propagation).